src/main/java/org/opensrp/web/GzipBodyDecompressFilter.java

```diff
@@ -130,7 +124,23 @@ public GzippedInputStreamWrapper(final HttpServletRequest request) throws IOExce
 		public ServletInputStream getInputStream() throws IOException {
 			final ByteArrayInputStream sourceStream = new ByteArrayInputStream(bytes);
 			return new ServletInputStream() {
-				
+				private ReadListener readListener;
+
+				@Override
+				public boolean isFinished() {
+					return false;
```

Comment:
This might probably crash or never finish reading. Have you tested the G-Zipping functionality with the server?

https://www.baeldung.com/spring-reading-httpservletrequest-multiple-times#servletInputStream-isFinished

I think it should be something similar to the explanation above. 

Alternatively, something like this in

```
    public int read() throws IOException {
        int data = this.sourceStream.read();
        if (data == -1) {
            this.finished = true;
        }

        return data;
    }
```

Reply:
makes sense